### PR TITLE
Warn users if they forgot to call wpilib.run

### PR DESCRIPTION
**Pull request by virtuald**
* Inspired by https://www.chiefdelphi.com/t/robot-code-not-deploying-from-robot-py/343672

### CODE REVIEW

File: wpilib/wpilib/_impl/main.py [hunk 1]
```diff
@@ -11,6 +12,31 @@
 from .logconfig import configure_logging
 import hal_impl
 
+# Make silly mistakes more obvious
+_show_run_warning = True
+
+
+def _atexit():
+    if _show_run_warning and not "pytest" in sys.modules:
```
Comment (auscompgeek) [inline wpilib/wpilib/_impl/main.py]: lol. perhaps it'd be a better idea to move the pytest check down to the atexit register call?

File: wpilib/wpilib/_impl/main.py [hunk 2]
```diff
@@ -11,6 +12,31 @@
...
+        print("")
+        print('    if __name__ == "__main__":')
+        print("        wpilib.run(MyRobot)")
+        print()
+
+
+def _excepthook(*args):
+    global _show_run_warning
+    _show_run_warning = False
+    _orig_excepthook(*args)
```
Comment (auscompgeek) [inline wpilib/wpilib/_impl/main.py]: maybe restore the original excepthook here too?

**Review by auscompgeek — COMMENTED** [review]
how does this interact with someone playing around with wpilib in a repl?

File: wpilib/wpilib/_impl/main.py (hunk 1 above)
Comment (virtuald) [reply wpilib/wpilib/_impl/main.py]: Eh.

**Comment by virtuald** [comment]
One question is how poorly this would interact with something doing inspection of wpilib for autocomplete et al.

File: wpilib/wpilib/_impl/main.py (hunk 1 above)
Comment (auscompgeek) [reply wpilib/wpilib/_impl/main.py]: Wouldn't this have a not-good impact on the time to run the tests though? Like, we have almost 900 tests, and wpilib gets reloaded for every one of those.

**Pull request closed without merging**